go/private/rules/binary.bzl

```diff
@@ -71,7 +71,7 @@ def _go_binary_impl(ctx):
     """go_binary_impl emits actions for compiling and linking a go executable."""
     go = go_context(ctx)
 
-    library = go.new_library(go, importable = False)
+    library = go.new_library(go, importable = False, importpath = "main")
```

Comment:
This isn't quite right.. The providers we return need to have the original import paths. `go_path` uses these to place source files within archives. Anything else that looks at providers or runs an aspect over the dependency graph will need this, too.

Perhaps instead, let's have an optional `is_main` parameter on `new_library`, which will set `is_main` within `GoLibrary`. `effective_importpath_packagepath` should return `"main"` for the package path if that flag is set (the import path should stay the same). We should use the result of `effective_importpath_packagepath` when calling `emit_compilepkg` in `emit_archive`.